nvidia-dnf.py

```diff
@@ -70,19 +78,30 @@ def sack(self):
             if ver_cmp_pkgs(sack, kernelpkg, installed_kernel) != 1:
                 continue
 
-            k_corepkg = str(kernelpkg).replace(KERNEL_PKG_NAME + "-", KERNEL_PKG_REAL + "-")
+            # Matching kernel-core package
+            try:
+                k_corepkg = [i for i in available_k_cores if i.version == kernelpkg.version and i.release == kernelpkg.release][0]
+            except:
+                print('Unable to find matching ' + KERNEL_PKG_REAL + ' package')
 
+            # Get package name
             kmod_pkg_name = KMOD_PKG_PREFIX + '-' + str(driver.version) + '-' + \
                     str(kernelpkg.version) + '-' + str(remove_release_dist(kernelpkg.release))
 
             kmod_pkg = sack.query().available().filter(name = kmod_pkg_name, version = driver.version)
             if not kmod_pkg:
-                # Exclude kernel
-                sack.add_excludes([kernelpkg])
-                sack.add_excludes([k_corepkg])
-                print('NOTE: Skipping kernel installation since no NVIDIA driver kernel module package ' + \
-                    str(kmod_pkg_name) + ' for kernel ' + str(kernelpkg) + ' and driver ' + \
-                    str(driver) + ' could be found')
+                # Exclude kernel packages
+                try:
+                    sack.add_excludes([kernelpkg])
+                    sack.add_excludes([k_corepkg])
+                    exclude = True
+                except Exception as error:
+                    print('WARNING: kernel exclude error', error)
+
+                if exclude == True:
+                    print('NOTE: Skipping kernel installation since no NVIDIA driver kernel module package ' + \
+                        str(kmod_pkg_name) + ' for kernel ' + str(kernelpkg) + ' and driver ' + \
+                        str(driver) + ' could be found')
```

Comment:
Can this print also go into the `try` block at the end and replace the `exclude` variable? Or is that not how Python's try/except works? I found it confusing that `exclude` is only defined in the `try` block but I guess that works in Pyhton.

Reply:
Fixed in #4